python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/dlt/resource.py

```diff
@@ -67,6 +69,23 @@ def _recursive_cast(value: Any):
 
         return {k: _recursive_cast(v) for k, v in mapping.items()}
 
+    def _extract_table_schema_metadata(self, table_name: str, schema: Schema) -> TableSchema:
+        # Pyright does not detect the default value from 'pop' and 'get'
+        return TableSchema(
+            columns=[
+                TableColumn(
+                    name=column.pop("name", ""),  # type: ignore
+                    type=column.pop("data_type", "string"),  # type: ignore
```

Comment:
Do we expect columns to exist without a `name` and a `data_type`? If that's the case, we could have multiple `TableColumn` objects with the empty string as a name?

Reply:
It shouldn't be possible because we are using dlt's `Schema.get_table_columns()` with the `include_incomplete` argument set to `False`. That ensures we only retrieve columns having a `name` and `data_type`.

I added default string values mostly for Pyright to be happy since `name` and `type` are nonoptional arguments in `TableColumn`. It should also prevent a key error in case `name` or `data_type` really are missing. My thinking was that a column having no `name` should be an empty string and a missing `data_type` most likely means an all `null` column (at least in Snowflake, a column such as `NULL as col1` is a `VARCHAR` by default). Choosing `string` seems like the safest option as it is also the default value for `TableColumn.type`.

I would argue it's an improvement over the old implementation which was using `column.get("name")`. This could raise an exception if `name` is missing since the `get()` method defaults to returning `None`.